pkg/node/node_test.go

```diff
@@ -106,14 +112,49 @@ func TestDrainSuccess(t *testing.T) {
 		},
 		metav1.CreateOptions{})
 	h.Ok(t, err)
+	_, err = client.CoreV1().Pods("default").Create(
+		context.Background(),
+		&v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "cool-app-pod-",
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: "apps/v1",
+						Name:       "cool-app",
+						Kind:       "ReplicaSet",
+						Controller: &controllerBool,
+					},
+				},
+			},
+			Spec: v1.PodSpec{
+				NodeName: nodeName,
+			},
+		},
+		metav1.CreateOptions{})
+	h.Ok(t, err)
+
+	fakeRecorder := record.NewFakeRecorder(100)
+
 	tNode := getNode(t, getDrainHelper(client))
-	err = tNode.CordonAndDrain(nodeName, "cordonReason")
+	err = tNode.CordonAndDrain(nodeName, "cordonReason", fakeRecorder)
+
 	h.Ok(t, err)
+	close(fakeRecorder.Events)
```

Comment:
@brycahta asked https://github.com/aws/aws-node-termination-handler/pull/631/files#r868327548

> Is there a reason to explicitly `close` this channel? If so, please update and use `defer` in case `h.Ok` fails.

If the channel is not closed, the upcoming `for event := range fakeRecorder.Events` loops forever. And if we use `defer` here, the test panics because `defer` attempts to close an already closed channel.

I have added two `defer` in the other two unit tests where the channel is created and never closed nor read. In this case here I agree that a failure in `h.Ok` would leave the channel open so I'm closing it before the assertion.